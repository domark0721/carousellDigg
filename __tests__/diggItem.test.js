import React from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import DiggItem from '../src/components/diggItem'

describe('component: DiggItem', () => {
    it('should render the DiggItem correctly', () => {
        const tree = renderer.create(
            <DiggItem
                id={0}
                topic='Test Digg Item Topic'
                upvoteCount={10}
                downvoteCount={20}
                onVoteDigg={() => {}}
            />,
        ).toJSON()

        // expect to match the snapshot
        expect(tree).toMatchSnapshot()
    })

    it('should callback when clicked upvote/downvote button', () => {
        const mockCallBack = jest.fn()
        const elDiggItem = shallow(
            <DiggItem
                id={0}
                topic='Test Digg Item Topic'
                upvoteCount={10}
                downvoteCount={20}
                onVoteDigg={mockCallBack}
            />,
        )

        // simulate click upvote button
        elDiggItem.find('.digg-button.upvote-button').simulate('click')

        // simulate click downvote button-label
        elDiggItem.find('.digg-button.downvote-button').simulate('click')

        expect(mockCallBack.mock.calls.length).toBe(2)

        // callback arguments for two clicks
        expect(mockCallBack.mock.calls[0][0]).toBe(0)
        expect(mockCallBack.mock.calls[0][1]).toBe('UP')
        expect(mockCallBack.mock.calls[1][0]).toBe(0)
        expect(mockCallBack.mock.calls[1][1]).toBe('DOWN')
    })
})
