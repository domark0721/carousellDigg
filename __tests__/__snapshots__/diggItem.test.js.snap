// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: DiggItem should render the DiggItem correctly 1`] = `
<div
  className="digg-item"
>
  <div
    className="digg-id"
  >
    #
    0
  </div>
  <p
    className="digg-topic"
  >
    Test Digg Item Topic
  </p>
  <div
    className="digg-buttons-wrap"
  >
    <div
      className="digg-button upvote-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="button-label"
      >
        Upvote
      </span>
      <span
        className="button-count"
      >
        10
      </span>
    </div>
    <div
      className="digg-button downvote-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="button-label"
      >
        Downvote
      </span>
      <span
        className="button-count"
      >
        20
      </span>
    </div>
  </div>
</div>
`;
